diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2016-10-02 09:57:17.000000000 +0800
+++ b/configure.ac	2016-10-02 10:22:00.771779764 +0800
@@ -26,6 +26,9 @@
 AC_CHECK_FUNCS([asprintf])
 AC_CHECK_LIB([m],[exp])
 
+## Checks for header files.
+AC_CHECK_HEADERS([error.h])
+
 # Debugging
 AC_ARG_ENABLE(debug,
         [  --enable-debug          enable debugging],,
diff -Naur a/src/webserver.c b/src/webserver.c
--- a/src/webserver.c	2015-05-07 11:57:30.000000000 +0800
+++ b/src/webserver.c	2016-10-02 10:34:51.932528036 +0800
@@ -31,7 +31,14 @@
 #include <sys/stat.h>
 #include <unistd.h>
 #include <errno.h>
+
+#if defined(HAVE_ERROR_H)
 #include <error.h>
+#else
+#include <err.h>
+#define error(status, errno, ...) err(status, __VA_ARGS__)
+#endif
+
 #include <string.h>
 #include <limits.h>
 #include <assert.h>
